### DISCUSSION TITLE
How to copy files into a podman machine

### DISCUSSION THREAD

**Discussion by prkadalbcisco** [containers/podman#17149] [General]
Hello everyone,

I installed podman today and I'm able to ssh into the podman machine `podman machine ssh`.

Is there any way to copy files into this machine from the host?

The podman cp is for container-to-host copying, but I didn't find anything for the machine-to-host copying.

Thank you

**Comment by Luap99** [comment]
You can get the ssh address from the `podman system connection list` output and then just use regular `scp` or `rsync` to copy, or you just use the mounted $HOME volume that is mounted into the VM by default (assuming you use MacOS or Linux). On windows you can access all host files via `/mnt/<DRIVE_LETTER>/...`

**Reply by maxx27** [reply]
Where can I find the credentials and/or key file for connection?

**Reply by r-mettner-skaylink** [reply]
dunno about windows but on mac,  the output of `podman system connection list` there is a row with Identity where the SSH keys are listed. 
technically you can also add your own ssh keys to the core users authorized keys file.

**Comment by rquinio1A** [comment]
The following works:

`podman machine ssh sh -c 'cat > //etc/containers/registries.conf.d/myconfig.conf' < myconfig.conf`

**Comment by amaddio** [comment]
For the sake of completeness and to save others some time, here would be the adequate scp solution:

```sh
scp -P <port> -R -i <identity_file> <source_folder> root@127.0.0.1:<target_folder>
```

Replace `<port>` and `<identity_file>` with values received from: `podman system connection list`

**Reply by mattisking** [reply]
thank you

**Comment by Luap99** [comment]
Note that `podman machine cp` was merged recently https://github.com/containers/podman/pull/25331 and thus will be part of podman v5.5.

**Comment by akhilgentoo** [comment]
We can also put permission to write to desired folder
example : podman machine ssh "chmod 777 /home"
